resolver/resolver.cc

```diff
@@ -1309,24 +1309,23 @@ class ResolveConstantsWalk {
             return false;
         }
 
-        if (nesting_->scope == core::Symbols::root()) {
+        if (curSym == core::Symbols::root()) {
             // no need to check <root> def itself
             return false;
         }
 
         // Can't be ambiguous if current definition is single-part, don't check in this case.
-        const core::SymbolRef curOwner = nesting_->scope.owner(ctx);
-        if (curOwner == nesting_->parent->scope) {
+        const core::SymbolRef curOwner = curSym.owner(ctx);
+        if (curOwner == curNesting->scope) {
             return false;
         }
 
         return true;
     }
 
-    const core::SymbolRef findAnyDefinitionAmbiguousWithCurrent(core::Context ctx) {
+    const core::SymbolRef findAnyDefinitionAmbiguousWithCurrent(core::Context ctx, core::SymbolRef curSym,
+                                                                shared_ptr<Nesting> curNesting) {
```

Comment:
Same comment here, though I think taking by `const &` means that we need to rewrite the loop in this function a little further down...which is probably not a bad idea anyway.

```suggestion
    const core::SymbolRef findAnyDefinitionAmbiguousWithCurrent(core::Context ctx, core::SymbolRef curSym,
                                                                const shared_ptr<Nesting> &curNesting) {
```